Sources/SPMBuildCore/BuildParameters/BuildParameters.swift

```diff
@@ -26,6 +26,18 @@ public struct BuildParameters: Encodable {
         case auto
     }
 
+    /// The destination for which code should be compiled for.
+    public enum Destination: Encodable {
```

Comment:
I think this is an intermediate type that currently connects to `buildTriple` for each resolved product/module. As a follow up to these changes I'm planning to remove `buildTriple`, this could only be accomplished if a build plan could be formed earlier than today and all of the "destination" handling would be moved there. Modules graph would have destination-less products/modules and build plan would be responsible for determining destinations and building complete set of host/target build descriptions that other types can query if they need a product/module for a particular destination. I major obstacle to this "plan" is how plugins are built/invoked - they create a separate build command and subsequently a new build plan etc. I'm hoping to change that and make `plugins/` directory obsolete.